docs/reference/modules/asset.md

```diff
@@ -0,0 +1,205 @@
+---
+extends: '@apostrophecms/module'
+---
+
+# `@apostrophecms/asset`
+
+**Alias:** `apos.asset`
+
+<AposRefExtends :module="$frontmatter.extends" />
+
+The `asset` module serves to organize, process, and output all project script and styling assets during the build process. In addition, it provides access to modify the project webpack configuration and exposes two CLI tasks for project building and webpack cache clearing. Options are passed through the creation of a `modules/@apostrophecms/asset/index.js` file.
+
+## Options
+
+|  Property | Type | Description |
+|---|---|---|
+| [`refreshOnRestart`](#refreshonrestart) | Boolean | If set to `true`, the browser will refresh on Apostrophe app restart. |
+| [`watch`](#watch) | Boolean | If set to `false`, none of the UI assets will be watched for triggering a restart. |
+| `watchDebounceMs` | Integer | Time in milliseconds to wait before re-triggering a restart on asset change. |
+
+### `refreshOnRestart`
+
+If `refreshOnRestart` is truthy and `process.env.NODE_ENV` is not set to `production`, then restart of the Apostrophe app will trigger a browser refresh.
+
+### `watch`
+
+If `watch` is falsy, then changes to any of the assets in the `modules` and `node_modules` will not trigger a restart. Note that this option is disabled and has no impact if `process.env.NODE_ENV` is set to `production`. 
+
+## Command Line Tasks
+
+### `build`
+The build command triggers the compilation, processing, and output of the project assets. The main code from each module, code placed in the `<module-name>/ui/src` folders, and scripts included in [bundles](#bundles) are combined separately. Any SCSS files are combined and translated to CSS. If `NODE_ENV=production`, all of the final deployment files are written out to the `public/releases` folder within a sub-folder named by the release id. The `build` script will try to automatically set the release id when building from a simple git checkout from the hash, building for Heroku or platform.sh from the environment variables, or Stagecoach from the root directory name. Otherwise, the release id can set from a `APOS_RELEASE_ID` environment variable or with a root directory file named 'release-id' with the value of the id inside. The value should be a short, unique string identifier.
+
+#### Example
+
+<AposCodeBlock>
+
+```sh
+APOS_RELEASE_ID='2022-01-01' node app @apostrophecms/asset:build
+```
+</AposCodeBlock>
+
+### `clear-cache`
+Some changes to the webpack configuration will not be detected automatically and will cause incorrect output. While this is rare, the caches for webpack can be emptied from the command line using the `clear-cache` task.
+
+#### Example
+
+<AposCodeBlock>
+
+```sh
+node app @apostrophecms/asset:clear-cache
+```
+</AposCodeBlock>
+
+## Webpack configuration
+
+The webpack configuration can be extended from any module. The `asset` module exposes three different properties for accomplishing this.
```

Comment:
This functionality extends the `ui/src` build, used for public-facing pages. The Apostrophe admin UI build configuration is fixed.

Reply:
Changed.